src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -559,56 +559,38 @@ internal bool CanNodeToCode(object parameters)
 
         internal void SelectInRegion(Rect2D region, bool isCrossSelect)
         {
-            bool fullyEnclosed = !isCrossSelect;
+            var fullyEnclosed = !isCrossSelect;
+            var selection = DynamoSelection.Instance.Selection;
+            var childlessModels = Model.Nodes.Concat<ModelBase>(Model.Notes);
 
-            foreach (NodeModel n in Model.Nodes)
+            foreach (var n in childlessModels)
             {
-                double x0 = n.X;
-                double y0 = n.Y;
-
                 if (IsInRegion(region, n, fullyEnclosed))
                 {
-                    if (!DynamoSelection.Instance.Selection.Contains(n))
-                        DynamoSelection.Instance.Selection.Add(n);
+                    selection.AddUnique(n);
                 }
                 else
                 {
                     if (n.IsSelected)
-                        DynamoSelection.Instance.Selection.Remove(n);
-                }
-            }
-
-            foreach (var n in Model.Notes)
-            {
-                double x0 = n.X;
-                double y0 = n.Y;
-
-                if (IsInRegion(region, n, fullyEnclosed))
-                {
-                    if (!DynamoSelection.Instance.Selection.Contains(n))
-                        DynamoSelection.Instance.Selection.Add(n);
-                }
-                else
-                {
-                    if (n.IsSelected)
-                        DynamoSelection.Instance.Selection.Remove(n);
+                        selection.Remove(n);
                 }
             }
 
             foreach (var n in Model.Annotations)
             {
-                double x0 = n.X;
-                double y0 = n.Y;
-
                 if (IsInRegion(region, n, fullyEnclosed))
                 {
-                    if (!DynamoSelection.Instance.Selection.Contains(n))
-                        DynamoSelection.Instance.Selection.Add(n);
+                    selection.AddUnique(n);
+                    // if annotation is selected its children should be added to selection too
+                    foreach (var m in n.SelectedModels)
+                    {
+                        selection.AddUnique(m);
+                    }
```

Comment:
![selection](https://cloud.githubusercontent.com/assets/7658189/10432558/83a378d0-7114-11e5-889f-ff99dc34e947.png)
In this case `Point` node visually selected but `Selection` doesn't contain it